src/test/kotlin/ru/kode/detekt/rule/compose/UnnecessaryLayoutWrapperTest.kt

```diff
@@ -0,0 +1,216 @@
+package ru.kode.detekt.rule.compose
+
+import io.gitlab.arturbosch.detekt.test.lint
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.collections.shouldBeEmpty
+import io.kotest.matchers.collections.shouldHaveSize
+
+class UnnecessaryLayoutWrapperTest : ShouldSpec({
+  should("not report when parent layout with any parameter has single child layout") {
+    // language=kotlin
+    val code = """
+      @Composable
+      fun Test() {
+        Row(modifier = Modifier.size(200.dp)) {
+          Column {
+      
+          }
+        }
+      }
+    """.trimIndent()
+
+    val findings = UnnecessaryLayoutWrapper().lint(code)
+
+    findings.shouldBeEmpty()
+  }
+
+  should("not report when child layout with any parameter has its own single child layout") {
+    // language=kotlin
+    val code = """
+      @Composable
+      fun Test() {
+        Box(modifier = Modifier.size(200.dp)) {
+          Box(modifier = Modifier.size(200.dp)) {
+            Row {
+      
+            }
+          }
+        }
+      }
+    """.trimIndent()
+
+    val findings = UnnecessaryLayoutWrapper().lint(code)
+
+    findings.shouldBeEmpty()
+  }
+
+  should("not report when parent parameterless layout has 2 or more child layouts") {
+    // language=kotlin
+    val code = """
+      @Composable
+      fun Test() {
+        Box {
+          Box {
+      
+          }
+          Column {
+      
+          }
+        }
+      }
+    """.trimIndent()
+
+    val findings = UnnecessaryLayoutWrapper().lint(code)
+
+    findings.shouldBeEmpty()
+  }
+
+  should("report when parent parameterless layout has single child layout") {
+    // language=kotlin
+    val code = """
+      @Composable
+      fun Test() {
+        Box {
```

Comment:
you should test all supported layout types here. data-driven testing feature can be used here:

https://kotest.io/docs/framework/datatesting/data-driven-testing.html